src/pynwb/base.py

```diff
@@ -159,6 +159,8 @@ def __init__(self, **kwargs):
         timestamps = kwargs.get('timestamps')
         starting_time = kwargs.get('starting_time')
         rate = kwargs.get('rate')
+        if rate is not None and timestamps is not None:
+            raise ValueError('specify rate[+starting_time] OR timestamps, not both')
```

Comment:
@bendichter How about checking starting_time as well?

```
if timestamps is not None:
    if rate is not None:
         raise ValueError('Specifying rate and timestamps is not supported.')
    elif starting_time is not None:
         raise ValueError('Specifying starting_time and timestamps is not supported.')
```

and then we need one more test as well.

Reply:
@bendichter And if we code it up like that we should also change the schema as https://github.com/NeurodataWithoutBorders/pynwb/blob/5ac6603fb4e9c570dd1d32a627138d75729e7f89/src/pynwb/data/nwb.base.yaml#L139 says that rate is required.

Reply:
Now that I look at it again, the format of the HDF5 is that `starting_time` is a value and `rate` is a mandatory attribute. I think the logic was so that `starting_time` and `rate` go hand-in-hand. I would personally prefer it to be reversed and have `starting_time` be an attribute of `rate`, since `starting_time` is usually `0.0` and unimportant and `rate` is always important. So I believe `rate` is intentionally required here.